cmd/kubectl_token.go

```diff
@@ -381,18 +379,26 @@ func basicAuth(input *LoginInput) (managementClient.Token, error) {
 		responseType = fmt.Sprintf("%s_%s", responseType, input.clusterID)
 	}
 
-	body := fmt.Sprintf(`{"responseType":%q, "username":%q, "password":%q}`, responseType, username, password)
+	reqBody := fmt.Sprintf(`{"responseType":%q, "username":%q, "password":%q}`, responseType, username, password)
 
-	url := fmt.Sprintf("%s/v3-public/%ss/%s?action=login", input.server, input.authProvider,
+	loginURL := fmt.Sprintf("%s/v3-public/%ss/%s?action=login", input.server, input.authProvider,
 		strings.ToLower(strings.Replace(input.authProvider, "Provider", "", 1)))
 
-	response, err := request(http.MethodPost, url, bytes.NewBufferString(body))
+	req, err := http.NewRequest(http.MethodPost, loginURL, bytes.NewBufferString(reqBody))
+	if err != nil {
+		return token, fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, respBody, err := doRequest(client, req)
+	if err == nil && resp.StatusCode != http.StatusOK {
+		err = fmt.Errorf("unexpected http status code %d", resp.StatusCode)
+	}
 	if err != nil {
 		return token, nil
 	}
 
 	apiError := map[string]interface{}{}
-	err = json.Unmarshal(response, &apiError)
+	err = json.Unmarshal(respBody, &apiError)
 	if err != nil {
 		return token, err
```

Comment:
Is it ok to return both error and token here ? Should it be the error alone instead, without the token ?

Reply:
TL;DR Yes, it's. OK.
The `token` is of the value `managementClient.Token` not the reference type `*managementClient.Token` in this case. The returned value is ignored at the call site L338 in the case of an error as it should be.